runtime/tests/Test_LightningMeasures.cpp

```diff
@@ -793,6 +1053,30 @@ TEMPLATE_LIST_TEST_CASE("Var(Tensor(HermitianObs, Hamiltonian()) test", "[Measur
     CHECK(sim->Var(ten) == Approx(0.8).margin(1e-5));
 }
 
+// TEMPLATE_LIST_TEST_CASE("Var(Tensor(HermitianObs, Hamiltonian()) shots test", "[Measures]",
+//                         SimTypes)
+// {
+//     std::unique_ptr<TestType> sim = std::make_unique<TestType>();
+
+//     // state-vector with #qubits = n
+//     constexpr size_t n = 3;
+//     std::vector<QubitIdType> Qs = sim->AllocateQubits(n);
+
+//     constexpr size_t num_shots = 5000;
+//     sim->SetDeviceShots(num_shots);
+
+//     sim->NamedOperation("Hadamard", {}, {Qs[0]}, false);
+//     sim->NamedOperation("PauliY", {}, {Qs[1]}, false);
+
+//     ObsIdType px = sim->Observable(ObsId::PauliX, {}, {Qs[1]});
+//     ObsIdType py = sim->Observable(ObsId::PauliY, {}, {Qs[2]});
+//     ObsIdType hxy = sim->HamiltonianObservable({0.4, 0.8}, {px, py});
+//     ObsIdType ten = sim->TensorObservable({hxy});
+
+//     CHECK(sim->Var(ten) == Approx(0.8).margin(5e-2)); // TODO: Lightning failed!,
```

Comment:
I've added one more test with the same issue while working on the support.. this should be uncommented before merge too.